packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -230,7 +228,8 @@ class ChangeNotifier implements Listenable {
   ///    changes.
   @override
   void removeListener(VoidCallback listener) {
-    assert(_debugAssertNotDisposed());
+    if (_debugDisposed)
```

Comment:
`_debug` variables should only be sued in debug contexts (i.e. other debug methods or asserts).

Reply:
good catch i totally forgot this